lib/commands/performance.js

```diff
@@ -0,0 +1,339 @@
+        return returnValue;
+      }
+    }
+    await sleep(1000);
+  }
+};
+
+helpers.getBatteryInfo = async function (dataReadTimeout) {
+  let returnValue = [];
+  returnValue[0] = ["power"];
+  let cmd, data, start, end, power;
+  let loop = true;
+  let tryCount = 0;
+  cmd = ['dumpsys', 'battery', '|', 'grep', "\'level\'"];
+
+  while ( loop ) {
+    if (tryCount === (dataReadTimeout + 1))
+      throw new Error('No response received after ' + dataReadTimeout);
+    else {
+      ++ tryCount;
+      data = await this.adb.shell(cmd);        
+    }
+
+    if (data !== "" && data !== undefined && data !== null){
+      start = data.indexOf(":");
+      end = data.length;
+      power = data.substring(start+1, end).trim();
+
+      if (power !== "" && power !== undefined && power !== "nodex"){
+        returnValue[1] = [power];
+        return returnValue;
+      }            
+    }
+    await sleep(1000);
+  }
+};
+
+helpers.getMemoryInfo = async function (packageName, dataReadTimeout) {
+  let returnValue = [];
+  returnValue[0] = ["totalPrivateDirty", "nativePrivateDirty", "dalvikPrivateDirty", "eglPrivateDirty", "glPrivateDirty", "totalPss", "nativePss", "dalvikPss", "eglPss", "glPss", "nativeHeapAllocatedSize", "nativeHeapSize"];
+    
+  let loop = true;
+  let cmd, data, totalPrivateDirty, nativePrivateDirty, dalvikPrivateDirty, eglPrivateDirty, glPrivateDirty, totalPss, nativePss, dalvikPss, eglPss, glPss, nativeHeapSize, nativeHeapAllocatedSize;
+  let tryCount = 0;
+      
+  cmd = ['dumpsys', 'meminfo', `'${packageName}'`, '|', 'grep', '-E', "\'Native|Dalvik|EGL|GL|TOTAL\'"];
+
+  while ( loop ) {    
+    if (tryCount === (dataReadTimeout + 1))
+      throw new Error('No response received after ' + dataReadTimeout);
+    else {
+      ++ tryCount;
+      data = await this.adb.shell(cmd);        
+    }
+        
+    if (data !== "" && data !== undefined && data !== null){
+      let arrayList = data.split("\n");
+      let arrayList2;  
+      for (let i=0 ; i < arrayList.length ; i++){
+        let testString = arrayList[i].replace(/\s/g, ","); // 문자열 끝 부분의 공백 제거
+        testString = testString.replace(/,{1,}/g, ","); // 문자열 끝 부분의 공백 제거
+        arrayList2 = testString.split(",");
+
+        if ((arrayList2[0] === "Native" || arrayList2[1] === "Native") && (arrayList2[1] === "Heap" || arrayList2[2] === "Heap")){//native heap
+          nativePrivateDirty = arrayList2[3] ;
+          nativePss = arrayList2[2];
+          nativeHeapAllocatedSize = arrayList2[6];
+          nativeHeapSize = arrayList2[8];
+        } else if ((arrayList2[0] === "Dalvik" || arrayList2[1] === "Dalvik") && (arrayList2[1] === "Heap" || arrayList2[2] === "Heap")) { //dalvik heap
+          dalvikPrivateDirty = arrayList2[4];
+          dalvikPss = arrayList2[3];
+        } else if ( (arrayList2[0] === "Dalvik" || arrayList2[1] === "Dalvik") && (arrayList2[1] === "Other" || arrayList2[2] === "Other") ) ;//dalvik others
```

Comment:
braces here are a little confusing. our style is that every `if/else` statement should have braces; no one-line if/else